gputop-data/gputop.proto

```diff
@@ -134,6 +134,11 @@ message TracepointInfo
     required string sample_format = 2;
 }
 
+message HW_ID
+{
+    repeated uint64 ctx_hw_id = 1;
+    repeated uint64 vgpu_id = 2;
+}
```

Comment:
Does this mean you have the UI query every single vgpu, hw_id tuple one by one?
Sounds like you should return an array instead.

Reply:
Yes, I have the UI query each vgpu to the vgpu_id and ctx_hw_id array one by one. Beside of the vgpu_id and ctx_id, we also get the current vgpu number, which is defined as "n_ctx_hw_id" and "n_vgpu_id". 
In the function handle_get_features, the array notices[] = "RC6 power saving mode disabled" is sent through defining the "notices" and "n_notices". In the same way, we send the vgpu_id array, ctx_hw_id array,  n_ctx_hw_id and n_ctx_hw_id every time after receiving the query in the gputop-server. Then the vgpu_id and ctx_hw_id array can be resolved in the client.